Local Development + extra exports

jburrow opened the pull request:
The context to the PR is i am working on a time-series plugin which can render cpu, memory in addition to the other plugins you have. I am also want to render multiple flame-plugins are the same time for comparisions use-case. 

This PR is a small one that
 - Introduces a simple local-dev experience with autoreload from npm start
 - Exports the {class} + default export to make imports in typescript "nicer" import {classname} from "script"
 - Export all the types.ts - some were missing.
 - Ability to override plugin instance name to support multiple instances of same pluging (for comparision) i have additional changes locally.

Thanks for your time and consideration.

James

Smrtnyk commented on package.json:
```diff
@@ -41,7 +42,7 @@
     "@types/events": "^3.0.0",
     "@typescript-eslint/eslint-plugin": "^5.39.0",
-    "@typescript-eslint/parser": "^5.39.0",
+    "@typescript-eslint/parser": "^5.39.0",    
```
this whitespaces got added?

jburrow replied:
agreed - sorry + fixed

Smrtnyk commented on rollup.config.js:
```diff
@@ -1,40 +1,39 @@
+import cleaner from 'rollup-plugin-cleaner';
+import { template } from './example/src/template.js';
+import typescript from '@rollup/plugin-typescript';
+
```
this whitespace seems superfluous

jburrow replied:
agreed - sorry + fixed

jburrow replied:
This is just ";" missing from end of lines.

![image](https://user-images.githubusercontent.com/1444266/208416581-4adbd3b5-7e30-48f1-9954-d92979d92dd4.png)

jburrow replied:
removed from PR

Smrtnyk commented on src/flame-chart.ts:
```diff
@@ -8,6 +8,7 @@ import MarksPlugin from './plugins/marks-plugin';
 import { Colors, Data, Marks, Waterfall } from './types';
 import UIPlugin from './plugins/ui-plugin';
 
+
```
same here

jburrow replied:
agreed - sorry + fixed

Smrtnyk commented on src/flame-chart.ts:
```diff
@@ -34,7 +35,7 @@ export type FlameChartOptions = {
 const defaultSettings: FlameChartSettings = {};
 
-export default class FlameChart extends FlameChartContainer<FlameChartStyles> {
+export class FlameChart extends FlameChartContainer<FlameChartStyles> {
```
named export and default export of the same thing?
not sure how to feel about that

jburrow replied:
I can understand how it looks odd - but it makes for a nicer typescript api (IMHO). It is backwards compat, non-breaking.

I have a lot more chagnes pending - converting the example to typescript - and the code looks cleaner without using the default export.

Smrtnyk reviewed (commented):
I am not an owner of this repo
just a contributor with rights to merge stuff

A proper review can give you Nikolay

From my side, I noticed there is some unneeded noise with extra added lines in files
PR would be smaller without those

jburrow commented:
Do you need me to do anything else?

pyatyispyatil reviewed (changes requested):
It would be very nice to have named exports for all plugins. Otherwise it will be very inconvenient for our users.

jburrow commented:
I have updated all the plugins to have have named exports, and updated their usage in index.ts to use the named instances.
This change is non-breaking.

I am sorry to say i don't know how to resolve this conflicts with package.json

jburrow commented:
many thanks
